vulnerability-analyzer/src/main/java/org/dependencytrack/vulnanalyzer/processor/scanner/snyk/SnykProcessor.java

```diff
@@ -217,13 +218,18 @@ public void analyzeBatch(final List<RetryableRecord<String, ScanTask>> batch) {
             LOGGER.debug("No results were returned for coordinates {} (affecting {}/{} records in this batch)",
                     unmatchedRecords.getKey(), unmatchedRecords.getValue().size(), batch.size());
             for (final RetryableRecord<String, ScanTask> record : unmatchedRecords.getValue()) {
+                // Setting vulnerability rejected timestamp for records with empty data, to suppress them in apiserver.
+                var bomToBeSuppressed = Bom.newBuilder()
+                        .addVulnerabilities(Vulnerability.newBuilder()
+                                .setRejected(Timestamp.newBuilder().setSeconds(Instant.now().getEpochSecond()).build()).build());
```

Comment:
Is this a safe assumption to make here? The comment above reads:

> [...] there is no way for us to differentiate between "no vulnerabilities" and "unable to correlate".

Also, a vulnerability record without ID and source doesn't make sense to be rejected. My understanding was that Snyk still returned the ID, but no content for it. But this makes me think they don't return it at all anymore?

Reply:
No they don't send any data i.e. `"data": []`